spec/02-integration/18-hybrid_rpc/06-validate_deltas_spec.lua

```diff
@@ -0,0 +1,200 @@
+local helpers = require "spec.helpers"
+local txn = require "resty.lmdb.transaction"
+local declarative = require "kong.db.declarative"
+
+
+local insert_entity_for_txn = declarative.insert_entity_for_txn
+local validate_deltas = require("kong.clustering.services.sync.validate").validate_deltas
+
+
+local function lmdb_drop()
+  local t = txn.begin(512)
+  t:db_drop(false)
+  t:commit()
+end
+
+
+local function lmdb_insert(name, entity)
+  local t = txn.begin(512)
+  local res, err = insert_entity_for_txn(t, name, entity, nil)
+  if not res then
+    error("lmdb insert failed: " .. err)
+  end
+
+  local ok, err = t:commit()
+  if not ok then
+    error("lmdb t:commit() failed: " .. err)
+  end
+end
+
+
+-- insert into LMDB
+local function db_insert(bp, name, entity)
+  -- insert into dc blueprints
+  entity = bp[name]:insert(entity)
+
+  -- insert into LMDB
+  lmdb_insert(name, entity)
+
+  assert(kong.db[name]:select({id = entity.id}))
+
+  return entity
+end
+
+
+-- Cache the declarative_config to avoid the overhead of repeatedly executing
+-- the time-consuming chain:
+--   declarative.new_config -> declarative_config.load -> load_plugin_subschemas
+local cached_dc
+
+local function setup_bp()
+  -- reset lmdb
+  lmdb_drop()
+
+  -- init bp / db ( true for expand_foreigns)
+  local bp, db = helpers.get_db_utils("off", nil, nil, nil, nil, true)
+
+  -- init workspaces
+  local workspaces = require "kong.workspaces"
+  workspaces.upsert_default(db)
+
+  -- init declarative config
+  if not cached_dc then
+    local err
+    cached_dc, err = declarative.new_config(kong.configuration, nil, nil, true)
```

Comment:
Do we need the last `true`? or we should restore the param of it?
Or we could use a fake `kong.sync`?

Reply:
let me check